lib/ide/ast/impl/src/lib.rs

```diff
@@ -18,7 +18,13 @@ use serde::Serialize;
 use shapely::*;
 use uuid::Uuid;
 
+/// Any kind of node metadata used in visualization.
+#[derive(Debug,Clone,Default,Deserialize,Serialize)]
+pub struct Metadata();
```

Comment:
Likely this will become just another map, storing arbitrary metadata for users. Unless we want to have this entries "atomic" and rather use multimap for `IdMetadataMap`.
Anyway, it should be assumed that multiple independent entities will want to store/load metadata for given ID and they should not interfere with each another.